### PR TITLE
Output exit code

### LINKED ISSUE #7: intuitive action exit code
As it is currently, the action current exits with a successful code (`0`) even when the run_check.sh finds problems given by clang-tidy or clang-format.

### Solution 
toward the end of the script, add
```bash
EXIT_CODE=0
if [ "$PAYLOAD_FORMAT" != "" && "$PAYLOAD_TIDY" != "" ]
then
  EXIT_CODE=1
fi
exit($EXIT_CODE)
```

## Counter-argument
If the intention is to allow PRs to merge errors (ignore the warning from clang-*), then the user's workflow could handle the exit code instead of hard-coding it in the run_checks.sh. This option would require a output from this action:

```yml
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: shenxianpeng/cpp-linter-action@master
        id: linter
        with:
          style: file
          extensions: 'cpp'
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

      # If there are any comments, fail the check
      - if: steps.linter.outputs.checks-failed > 0
        run: exit 1
```

This would simply require a change to the above bash script suggestion
```bash
EXIT_CODE=0
if [ "$PAYLOAD_FORMAT" != "" && "$PAYLOAD_TIDY" != "" ]
then
  EXIT_CODE=1
fi
# exit($EXIT_CODE) # let action exit successfully

# set the action's `checks-failed` output
echo "::set-output name=checks-failed::$(echo $EXIT_CODE)"
```

### PR DESCRIPTION
**Pull request by 2bndy5**
resolves #7 
- added a step to the test.yml so it `echo`s the new output but the action doesn't fail if the checks fail.
- also used this chance to exit the run_checks.sh script early if no files need to be checked.
- switched Docker container to shenxianpeng/clang-tools which saves a significant amount of setup time
   - I also removed Dockerfile's CMD to install CMake (wasn't ever needed) and clang* tools that are now provided by docker image.

NOTE: the output code is automatically removed from the logs, so we won't actually see the text
```
::set-output name=checks-failed::<EXIT_CODE>
```

### CODE REVIEW

**Comment by shenxianpeng** [comment]
You don't need to sleep? have a good rest. I‘ll study this PR tonight.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -7,9 +7,18 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       # - uses: actions/checkout@v2
-      - uses: shenxianpeng/cpp-linter-action@master
+      # - uses: shenxianpeng/cpp-linter-action@master
+      - uses: 2bndy5/cpp-linter-action@output-exit-code
```
Comment (shenxianpeng) [inline .github/workflows/test.yml:11]: Do you need to revert this change before merging?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -7,9 +7,18 @@ jobs:
...
         with:
           style: file
           extensions: 'cpp'
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Fail fast?!
+        if: steps.linter.outputs.checks-failed > 0
+        run: |
+          echo "Some files failed the linging checks!"
```
Comment (shenxianpeng) [inline .github/workflows/test.yml:22]: linging? or linting

**Review by shenxianpeng — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (2bndy5) [reply .github/workflows/test.yml:11]: Yes I do. Good catch.
